### PR TITLE
Fix native mkbundle

### PR DESCRIPTION
**Pull request by directhex**
This mostly resolves the worst parts of https://github.com/mono/mono/issues/16991 (but does not deal with the issues in mkbundle itself when it comes to tracking native dependencies automatically).

Using a temporary directory rather than "more correct" use of a temporary file w/ rename, because `zipnote` on macOS is broken and cannot do renames.

### CODE REVIEW

File: scripts/mono-package-runtime [hunk 1]
```diff
@@ -35,6 +35,6 @@ fi
...
+(zip -u $o.zip bin/mono lib/mono/4.5/mscorlib.dll lib/mono/4.5/System*dll lib/mono/4.5/Mono.CSharp.dll lib/mono/4.5/Microsoft*dll lib/mono/4.5/FSharp*.dll lib/mono/4.5/I18N*dll lib/mono/4.5/Accessibility.dll lib/mono/4.5/RabbitMQ.Client.dll lib/mono/4.5/ICSharpCode.SharpZipLib.dll lib/mono/4.5/CustomMarshalers.dll etc/mono/4.5/machine.config etc/mono/4.5/web.config lib/mono/4.5/Mono.Cairo.dll lib/mono/4.5/Mono.Data.Sqlite.dll lib/mono/4.5/Mono.Posix.dll lib/mono/4.5/Mono.Security.*dll lib/mono/4.5/Mono.Simd.dll lib/mono/4.5/Mono.WebBrowser.dll lib/mono/4.5/Novell.Directory.Ldap.dll lib/libMonoSupportW$extension lib/libMonoPosixHelper$extension lib/libmono-btls-shared$extension lib/libmono-native$extension lib/libmono-native-compat$extension)
```
Comment (akoeplinger) [inline scripts/mono-package-runtime]: libmono-native-compat will only exist on macOS, is that going to be a problem?

Comment (directhex) [reply scripts/mono-package-runtime]: No, `zip` treats files it doesn't know about as a warning, not an error. This was the easiest way to make this line work on both Mac and Linux.

```
	zip warning: name not matched: lib/libmono-native.dylib
  adding: bin/mono (deflated 59%)
```

**Comment by directhex** [comment]
@monojenkins build failed

**Comment by directhex** [comment]
@monojenkins build failed

**Pull request merged**